### PR TITLE
Also bump CI matrix include jobs to latest stable CPython version.

### PR DESCRIPTION
**Pull request by adamreichold**

### CODE REVIEW

**Comment by adamreichold** [comment]
Not sure where those requirements are configured? Direclty in the GitHub UI?

grafik

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
       - name: Run cargo checks
         run: |
           set -x
-          VERSIONS=("3.7" "3.8" "3.9" "3.10")
+          VERSIONS=("3.7" "3.8" "3.9" "3.10" "3.10" "3.11")
```
Comment (davidhewitt) [inline .github/workflows/ci.yml:69]: Looks like 3.10 unintentionally duplicated

```suggestion
          VERSIONS=("3.7" "3.8" "3.9" "3.10" "3.11")
```

**Review by davidhewitt — COMMENTED** [review]
Thanks, yes we'll need to update in the GitHub UI just before merging this (and everything else then has to be rebased).

**Review by davidhewitt — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -409,7 +409,7 @@ jobs:
       - uses: actions/setup-python@v4
         id: setup-python
         with:
-          python-version: 3.11.0-beta.4
+          python-version: "3.11"
```
Comment (messense) [inline .github/workflows/ci.yml:412]: https://github.com/PyO3/pyo3/blob/6766d9f93bd2f4130953627d7dc6e8758acc8e16/emscripten/Makefile#L6

I think this also needs to be updated.

Comment (adamreichold) [reply .github/workflows/ci.yml:392]: Good catch! Removed the variable outright.

**Review by messense — APPROVED** [review]

**Comment by messense** [comment]
> Not sure where those requirements are configured? Direclty in the GitHub UI for protected branch rules?

Updated.

**Comment by davidhewitt** [comment]
Thanks both - my family and I were out of action sick for the past week, better now thankfully, just catching up on everything I missed!

**Pull request merged**